Source/WebCore/accessibility/AccessibilityNodeObject.cpp

```diff
@@ -440,8 +440,21 @@ AccessibilityRole AccessibilityNodeObject::determineAccessibilityRoleFromNode(Tr
         return AccessibilityRole::LandmarkMain;
     if (node()->hasTagName(navTag))
         return AccessibilityRole::LandmarkNavigation;
-    if (node()->hasTagName(asideTag))
+    // The aside element should not assume the complementary role when nested
+    // within the sectioning content elements.
+    // https://w3c.github.io/html-aam/#el-aside
+    if (node()->hasTagName(asideTag)) {
+        // Return LandmarkComplementary if the aside element explicitly has a role="complementary".
```

Comment:
I think this comment is unnecessary — the code does a good job describing the intention. What do you think about removing this?

Reply:
Okay, I will remove it.